ios/brave-ios/Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Paged/YoutubeQualityScriptHandler.swift

```diff
@@ -12,13 +12,24 @@ import WebKit
 
 class YoutubeQualityScriptHandler: NSObject, TabContentScript, TabObserver {
   private var url: URL?
-  private var urlObserver: NSObjectProtocol?
+  private var reachableObserver: NSObjectProtocol?
 
   init(tab: some TabState) {
     self.url = tab.visibleURL
     super.init()
 
     tab.addObserver(self)
+    Reach.startMonitoring()
+
+    reachableObserver = NotificationCenter.default.addObserver(
+      forName: .reachabilityStatusChanged,
+      object: nil,
+      queue: .main
+    ) { [weak tab] notification in
+      if let status = notification.userInfo?["status"] as? ReachabilityStatus, let tab = tab {
+        Self.handleConnectionStatusChanged(status, tab: tab)
+      }
+    }
```

Comment:
This shouldn't be here, even just passing in a tab is only there for legacy purposes, replace with a tab helper